ext/mbstring/mbstring.c

```diff
@@ -2941,6 +2941,144 @@ PHP_FUNCTION(mb_strtolower)
 	RETURN_STR(mbstring_convert_case(PHP_UNICODE_CASE_LOWER, ZSTR_VAL(str), ZSTR_LEN(str), enc));
 }
 
+typedef enum {
+	MB_LTRIM = 1,
+	MB_RTRIM = 2,
+	MB_BOTH_TRIM = 3
+} MB_TRIM_MODE;
+
+static bool is_trim_wchar(uint32_t w, HashTable *ht)
+{
+	return zend_hash_index_exists(ht, w);
+}
+
+static zend_string* trim_each_wchar(zend_string *str, zend_string *what, MB_TRIM_MODE mode, const mbfl_encoding *enc)
+{
+	unsigned char *in = (unsigned char*)ZSTR_VAL(str);
+	unsigned char *what_in = (unsigned char*)ZSTR_VAL(what);
+	uint32_t wchar_buf[128];
+	uint32_t what_wchar_buf[128];
+	size_t in_len = ZSTR_LEN(str);
+	size_t out_len = 0;
+	size_t what_out_len = 0;
+	size_t what_len = ZSTR_LEN(what);
+	unsigned int state = 0;
+	size_t left = 0;
+	size_t right = 0;
+	size_t total_len = 0;
+	bool first = true;
+
+	if (what_len == 0) {
+		return zend_string_copy(str);
+	}
+
+	HashTable what_ht;
+	zval val;
+	ZVAL_EMPTY_STRING(&val);
```

Comment:
```suggestion
	ZVAL_TRUE(&val);
```

Nit: I think this is a bit clearer, because string may be confusing in this context. (It's also one instruction less in assembly although that doesn't really matter here).